### PR TITLE
Added update and destroy user functions

### PR DESCRIPTION
**Pull request by eewilcox**

### CODE REVIEW

File: spec/features/05_user_updates_user_info_spec.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    fill_in 'Email', with: user.email
+    fill_in 'user_password', with: user.password
+    click_button 'Sign In'
+
+    click_link 'Change My User Info'
+    click_button 'Cancel my account'
+
+    expect(page).to have_content('Bye! Your account has been successfully cancelled. We hope to see you again soon.')
+  end
+
```
Comment (lilybarrett) [inline spec/features/05_user_updates_user_info_spec.rb]: These are great, super detailed tests! However, I'd like to see some tests for the "unhappy path," as well, i.e., ensuring unauthenticated users cannot update/delete their account details. Please add in some relevant tests for that edge case and let me know when the PR is updated. :)

File: spec/features/05_user_updates_user_info_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  scenario 'an unauthenticated user cannot navigate to user edit page (they will be redirected to sign in page)' do
+    user = FactoryGirl.create(:user)
+    visit edit_user_registration_path
+
+    expect(page).to have_content('You need to sign in or sign up before continuing.')
+    expect(page).to_not have_content('Update')
+    expect(page).to_not have_content('Cancel my account')
+  end
+
```
Comment (lilybarrett) [inline spec/features/05_user_updates_user_info_spec.rb]: Awesome! Great test.
